### ISSUE TITLE
SharePoint ADFS deployment error: The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program

### ISSUE THREAD

**Issue by joecoolish** [Azure/azure-quickstart-templates#8745]
[sharepoint-adfs](https://github.com/Azure/azure-quickstart-templates/tree/master/sharepoint-adfs)

### Repro steps
1. Click "Deploy to Azure" button
2. Use the following parameters:
   - location: eastus
   - sharePointVersion: 2019
   - dnsLabelPrefix: abc
   -  domainFQDN: abc.search
   - adminUserName: joel
   - adminPassword: <At least 2 lower, 2 upper, 2 numbers, 2 specials, and 16 chars long>
   - vmsAutoShutdownTime: 9999
   - enableHybridBenefitServerLicenses: Yes
3. Everything else is left as default
4. Wait for 70 mins
5. Receive error

```
{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "VMExtensionProvisioningError",
      "message": "VM has reported a failure when processing extension 'ConfigureSPVM'. Error message: \"DSC Configuration 'ConfigureSPVM' completed with error(s). Following are the first few: The PowerShell DSC resource '[cChocoInstaller]InstallChoco' with SourceInfo 'C:\\Packages\\Plugins\\Microsoft.Powershell.DSC\\2.83.0.0\\DSCWork\\ConfigureSPVM.0\\ConfigureSPVM.ps1::192::9::cChocoInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details. The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. The SendConfigurationApply function did not succeed. LCM failed to start desired state configuration manually.\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionDSCWindowsTroubleshoot "
    }
  ]
}
```

With text:

> VM has reported a failure when processing extension 'ConfigureSPVM'. Error message: "DSC Configuration 'ConfigureSPVM' completed with error(s). Following are the first few: The PowerShell DSC resource '[cChocoInstaller]InstallChoco' with SourceInfo 'C:\Packages\Plugins\Microsoft.Powershell.DSC\2.83.0.0\DSCWork\ConfigureSPVM.0\ConfigureSPVM.ps1::192::9::cChocoInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details. The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. The SendConfigurationApply function did not succeed. LCM failed to start desired state configuration manually."

> More information on troubleshooting is available at https://aka.ms/VMExtensionDSCWindowsTroubleshoot

**Comment by Yvand** [comment]
@joecoolish this deployment error is caused by [this Chocolatey issue](https://github.com/chocolatey/cChoco/issues/151) which was fixed in the [latest version](https://www.powershellgallery.com/packages/cChoco/2.5.0.0).
I submitted [this PR](https://github.com/Azure/azure-quickstart-templates/pull/8737) to include the updated package and fix the deployment error, I'm waiting for it to be approved.

**Comment by mohamedkofta** [comment]
> [Sharepoint-adfs](https://github.com/Azure/azure-quickstart-templates/tree/master/sharepoint-adfs)
> 
> ### تفاصيل القضية
> ### خطوات Repro
> 1. انقر فوق الزر "نشر في Azure"
> 2. استخدم المعلمات التالية:
>    
>    * الموقع: إيستوس
>    * الإصدار: 2019
>    * dnsLabelPrefix: abc
>    * domainFQDN: abc.search
>    * adminUserName: جويل
>    * adminPassword: <على الأقل رقمان سفليان ورقمان علويان ورقمان خاصان وطول 16 حرفًا>
>    * الوقت: 9999
>    * enableHybridBenefitServerLicenses: نعم
> 3. يتم ترك كل شيء آخر كافتراضي
> 4. انتظر 70 دقيقة
> 5. تلقي خطأ
> 
> ```
> {
>   "code": "DeploymentFailed",
>   "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
>   "details": [
>     {
>       "code": "VMExtensionProvisioningError",
>       "message": "VM has reported a failure when processing extension 'ConfigureSPVM'. Error message: \"DSC Configuration 'ConfigureSPVM' completed with error(s). Following are the first few: The PowerShell DSC resource '[cChocoInstaller]InstallChoco' with SourceInfo 'C:\\Packages\\Plugins\\Microsoft.Powershell.DSC\\2.83.0.0\\DSCWork\\ConfigureSPVM.0\\ConfigureSPVM.ps1::192::9::cChocoInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details. The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. The SendConfigurationApply function did not succeed. LCM failed to start desired state configuration manually.\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionDSCWindowsTroubleshoot "
>     }
>   ]
> }
> ```
> 
> مع النص:
> 
> > VM has reported a failure when processing extension 'ConfigureSPVM'. Error message: "DSC Configuration 'ConfigureSPVM' completed with error(s). Following are the first few: The PowerShell DSC resource '[cChocoInstaller]InstallChoco' with SourceInfo 'C:\Packages\Plugins\Microsoft.Powershell.DSC\2.83.0.0\DSCWork\ConfigureSPVM.0\ConfigureSPVM.ps1::192::9::cChocoInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details. The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. The SendConfigurationApply function did not succeed. LCM failed to start desired state configuration manually."
> 
> > More information on troubleshooting is available at https://aka.ms/VMExtensionDSCWindowsTroubleshoot

**Comment by mohamedkofta** [comment]
> [sharepoint-adfs](https://github.com/Azure/azure-quickstart-templates/tree/master/sharepoint-adfs)
> 
> ### Issue Details
> ### Repro steps
> 1. Click "Deploy to Azure" button
> 2. Use the following parameters:
>    
>    * location: eastus
>    * sharePointVersion: 2019
>    * dnsLabelPrefix: abc
>    * domainFQDN: abc.search
>    * adminUserName: joel
>    * adminPassword: <At least 2 lower, 2 upper, 2 numbers, 2 specials, and 16 chars long>
>    * vmsAutoShutdownTime: 9999
>    * enableHybridBenefitServerLicenses: Yes
> 3. Everything else is left as default
> 4. Wait for 70 mins
> 5. Receive error
> 
> ```
> {
>   "code": "DeploymentFailed",
>   "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
>   "details": [
>     {
>       "code": "VMExtensionProvisioningError",
>       "message": "VM has reported a failure when processing extension 'ConfigureSPVM'. Error message: \"DSC Configuration 'ConfigureSPVM' completed with error(s). Following are the first few: The PowerShell DSC resource '[cChocoInstaller]InstallChoco' with SourceInfo 'C:\\Packages\\Plugins\\Microsoft.Powershell.DSC\\2.83.0.0\\DSCWork\\ConfigureSPVM.0\\ConfigureSPVM.ps1::192::9::cChocoInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details. The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. The SendConfigurationApply function did not succeed. LCM failed to start desired state configuration manually.\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionDSCWindowsTroubleshoot "
>     }
>   ]
> }
> ```
> 
> With text:
> 
> > VM has reported a failure when processing extension 'ConfigureSPVM'. Error message: "DSC Configuration 'ConfigureSPVM' completed with error(s). Following are the first few: The PowerShell DSC resource '[cChocoInstaller]InstallChoco' with SourceInfo 'C:\Packages\Plugins\Microsoft.Powershell.DSC\2.83.0.0\DSCWork\ConfigureSPVM.0\ConfigureSPVM.ps1::192::9::cChocoInstaller' threw one or more non-terminating errors while running the Set-TargetResource functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details. The term 'Choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. The SendConfigurationApply function did not succeed. LCM failed to start desired state configuration manually."
> 
> > More information on troubleshooting is available at https://aka.ms/VMExtensionDSCWindowsTroubleshoot

**Comment by bmoore-msft** [comment]
should be fixed in #8737
